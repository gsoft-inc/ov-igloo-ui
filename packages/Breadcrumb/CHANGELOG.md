# @igloo-ui/breadcrumb

## 0.3.7

### Patch Changes

- 81723d0: Updated Hopper depedency
- Updated dependencies [81723d0]
  - @igloo-ui/provider@0.1.3

## 0.3.6

### Patch Changes

- 1613653: Updated Hopper Tokens Fonts package
- Updated dependencies [1613653]
  - @igloo-ui/provider@0.1.2

## 0.3.5

### Patch Changes

- 8281f80: Bumped dependency

## 0.3.4

### Patch Changes

- a99dfba: Fix UI issue

## 0.3.3

### Patch Changes

- 4a2812e: Added locale support using Igloo's provider.
- Updated dependencies [4a2812e]
  - @igloo-ui/provider@0.1.1

## 0.3.2

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper

## 0.3.1

### Patch Changes

- 42efe0b: Fix issue where classNames were not passed to children when cloned

## 0.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.2.5

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.

## 0.2.4

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.2.3

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.2.2

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.2.1

### Patch Changes

- 965c3d0: Fixed the double line for Igloo brand.

## 0.2.0

### Minor Changes

- ef3b3ca: Applied the rebrand to Breadcrumb.

## 0.1.5

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.1.4

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.

## 0.1.3

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.1.2

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.1

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2

## 0.1.0

### Minor Changes

- ed49b22: Feat: Initial release of the Breadcrumb component
