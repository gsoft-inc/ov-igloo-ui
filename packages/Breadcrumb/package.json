{
  "name": "@igloo-ui/breadcrumb",
  "version": "0.0.0",
  "main": "dist/Breadcrumb.js",
  "module": "dist/Breadcrumb.js",
  "types": "dist/Breadcrumb.d.ts",
  "styles": "dist/breadcrumb.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/tokens": "^1.0.0",
    "classnames": "^2.3.1",
    "@igloo-ui/icons": "^1.3.0"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
