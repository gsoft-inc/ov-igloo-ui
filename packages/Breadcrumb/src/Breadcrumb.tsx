import * as React from 'react';

import cx from 'classnames';

import ChevronLeft from '@igloo-ui/icons/dist/ChevronLeft';
import ChevronRight from '@igloo-ui/icons/dist/ChevronRight';

import './breadcrumb.scss';

type Item = {
  /** Add text for the breadcrumb */
  label: React.ReactNode;
  link?: string;
};

export interface BreadcrumbProps extends React.ComponentProps<'ol'> {
  /** Add a specific class to the tabs component */
  className?: string;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** Add a list of items to the breadcrumb */
  items: Item[];
}

const Breadcrumb: React.FunctionComponent<BreadcrumbProps> = ({
  className,
  dataTest,
  items,
}: BreadcrumbProps) => {
  const classes = cx('ids-breadcrumb', className);

  const listItems = items.map((item, index) => {
    const { label, link } = item;
    const current = !link && typeof label === 'string';
    const isCustomLabel = typeof label !== 'string';
    const isBackLink = items.length === 1 && (link || isCustomLabel);
    const itemClasses = cx('ids-breadcrumb__item', {
      'ids-breadcrumb__item--current': current,
      'ids-breadcrumb__item--is-back-link': isBackLink,
    });

    let displayLabel = label;
    if (isCustomLabel) {
      displayLabel = React.cloneElement(label as React.ReactElement, {
        className: 'ids-breadcrumb__link',
      });
    }

    return (
      <li
        key={`breadcrumb${index.toString()}`}
        aria-current={current && 'page'}
        className={itemClasses}
      >
        {isBackLink && (
          <ChevronLeft size="small" className="ids-breadcrumb__chevron" />
        )}
        {item.link ? (
          <a href={link} className="ids-breadcrumb__link">
            {displayLabel}
          </a>
        ) : (
          displayLabel
        )}

        {!isBackLink && !current && (
          <ChevronRight size="small" className="ids-breadcrumb__chevron" />
        )}
      </li>
    );
  });

  return (
    <nav aria-label="Breadcrumb" className={classes} data-test={dataTest}>
      <ol className="ids-breadcrumb__list">{listItems}</ol>
    </nav>
  );
};

export default Breadcrumb;
