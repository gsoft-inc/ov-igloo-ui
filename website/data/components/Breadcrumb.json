[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Breadcrumb/src/Breadcrumb.tsx",
      "description": "",
      "displayName": "Breadcrumb",
      "methods": [],
      "props": {
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the tabs component",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
            "name": "BreadcrumbProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
              "name": "BreadcrumbProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
            "name": "BreadcrumbProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
              "name": "BreadcrumbProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "items": {
          "defaultValue": null,
          "description": "Add a list of items to the breadcrumb",
          "name": "items",
          "parent": {
            "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
            "name": "BreadcrumbProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Breadcrumb/src/Breadcrumb.tsx",
              "name": "BreadcrumbProps"
            }
          ],
          "required": true,
          "type": { "name": "Item[]" }
        }
      }
    }
  ]
]
