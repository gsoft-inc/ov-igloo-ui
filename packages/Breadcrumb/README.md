# Breadcrumb

Breadcrumbs allow users to orient themselves in the hierarchy of the app. It also provides a way to navigate to the parent page.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/breadcrumb` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/breadcrumb
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/breadcrumb
```

## Usage

Then to use the component in your code just import it!

```jsx
import Breadcrumb from '@igloo-ui/breadcrumb';

<Breadcrumb
  items={[
    {
      label: 'Home',
      link: '#',
    },
    {
      label: 'Career',
      link: '#',
    },
    {
      label: 'Developer',
    },
  ]}
/>;
```
