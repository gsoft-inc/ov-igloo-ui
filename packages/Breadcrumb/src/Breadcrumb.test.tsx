/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Breadcrumb from './Breadcrumb';

describe('Breadcrumb', () => {
  test('It should render without error and a snapshot', () => {
    const { asFragment } = render(
      <Breadcrumb
        dataTest="breadcrumb1"
        items={[
          {
            label: 'Home',
            link: '#',
          },
          {
            label: 'Career',
            link: '#',
          },
          {
            label: 'Developer',
          },
        ]}
      />
    );
    expect(screen.getByTestId('breadcrumb1')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render a left pointing chevron', () => {
    const { container } = render(
      <Breadcrumb
        items={[
          {
            label: 'Back to Main page',
            link: '#',
          },
        ]}
      />
    );
    const leftChevron = container.querySelector(
      '.ids-breadcrumb__item--is-back-link .ids-breadcrumb__chevron'
    );
    expect(leftChevron).toBeInTheDocument();
  });
});
