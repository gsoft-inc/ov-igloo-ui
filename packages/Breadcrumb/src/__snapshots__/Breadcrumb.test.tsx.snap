// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb It should render without error and a snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    class="ids-breadcrumb"
    data-test="breadcrumb1"
  >
    <ol
      class="ids-breadcrumb__list"
    >
      <li
        aria-current="false"
        class="ids-breadcrumb__item"
      >
        <a
          class="ids-breadcrumb__link"
          href="#"
        >
          Home
        </a>
        <svg
          class="ids-icon ids-breadcrumb__chevron"
          fill="none"
          height="16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 4 3.823 3.823a.25.25 0 0 1 0 .354L6 12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </li>
      <li
        aria-current="false"
        class="ids-breadcrumb__item"
      >
        <a
          class="ids-breadcrumb__link"
          href="#"
        >
          Career
        </a>
        <svg
          class="ids-icon ids-breadcrumb__chevron"
          fill="none"
          height="16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 4 3.823 3.823a.25.25 0 0 1 0 .354L6 12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </li>
      <li
        aria-current="page"
        class="ids-breadcrumb__item ids-breadcrumb__item--current"
      >
        Developer
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
